prime-dotnet-webapi/Services/SiteRegistrationService.cs

```diff
@@ -0,0 +1,35 @@
+using Prime.Models.Api;
+using Prime.Engines;
+using Prime.Models;
+
+namespace Prime.Services
+{
+    public class SiteRegistrationService : ISiteRegistrationService
+    {
+        public bool CanPerformSiteStatusAction(SiteRegistrationAction action, SiteStatusType currentStatus)
+        {
+            SiteStatusType newStatus;
+
+            switch (action)
+            {
+                case SiteRegistrationAction.Submit:
+                case SiteRegistrationAction.Undecline:
+                    newStatus = SiteStatusType.InReview;
+                    break;
+                case SiteRegistrationAction.Approve:
+                    newStatus = SiteStatusType.Approved;
+                    break;
+                case SiteRegistrationAction.Decline:
+                    newStatus = SiteStatusType.Locked;
+                    break;
+                case SiteRegistrationAction.RequestChange:
+                case SiteRegistrationAction.Unapprove:
+                    newStatus = SiteStatusType.Active;
+                    break;
+                default:
+                    return false;
+            }
+            return SiteStatusStateEngine.AllowableStatusChange(currentStatus, newStatus);
```

Comment:
What is the purpose of having both this Service with `CanPerformSiteStatusAction()` and the `SiteStatusEngine` with `AllowableStatusChange()`? 
It feels like you only need this one method; the important question is can you perform the given action in the given status.

Reply:
> What is the purpose of having both this Service with `CanPerformSiteStatusAction()` and the `SiteStatusEngine` with `AllowableStatusChange()`?
> It feels like you only need this one method; the important question is can you perform the given action in the given status.

Can defintely merge them into one place. The original idea was just to make it the same pattern of the SubmissionStateEngine which also has a wrapping service layer.